Lib/typing.py

```diff
@@ -1393,7 +1393,11 @@ def __mro_entries__(self, bases):
                 return ()
             i = bases.index(self)
             for b in bases[i+1:]:
-                if isinstance(b, _BaseGenericAlias) and b is not self:
+                if b is self:
+                    continue
+                if isinstance(b, _BaseGenericAlias):
```

Comment:
Both of these diffs would make all tests pass:

```suggestion
                if isinstance(b, (_BaseGenericAlias, types.GenericAlias)):
```

```suggestion
                if isinstance(b, types.GenericAlias):
                    b = b.__origin__
                if isinstance(b, _BaseGenericAlias):
```

_But_, they can't both be right ;)